packages/core/src/sync-historical/service.ts

```diff
@@ -463,60 +518,21 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
         network: this.network.name,
       });
     }
-
-    this.queue.start();
   }
 
-  kill = async () => {
-    this.isKilling = true;
-    clearInterval(this.progressLogInterval);
-
-    this.queue.pause();
-    this.queue.clear();
-    await this.onIdle();
-
-    this.common.logger.debug({
-      service: "historical",
-      msg: `Killed historical sync service (network=${this.network.name})`,
-    });
-  };
-
-  onIdle = () =>
-    new Promise((resolve) =>
-      setImmediate(() => this.queue.onIdle().then(resolve)),
-    );
-
-  private buildQueue = () => {
-    const worker: Worker<HistoricalSyncTask> = async ({
-      task,
-      queue,
-      signal,
-    }) => {
-      switch (task.kind) {
-        case "LOG_FILTER": {
-          await this.logFilterTaskWorker({ task, signal });
-          break;
-        }
-        case "FACTORY_CHILD_ADDRESS": {
-          await this.factoryChildAddressTaskWorker({ task, signal });
-          break;
-        }
-        case "FACTORY_LOG_FILTER": {
-          await this.factoryLogFilterTaskWorker({ task, signal });
-          break;
-        }
-        case "BLOCK": {
-          await this.blockTaskWorker({ task, signal });
-          break;
-        }
-      }
-
-      // If this is not the final task, return.
-      if (queue.size > 0 || queue.pending > 1) return;
-      // If this is the final task but the kill() method has been called, do nothing.
-      if (this.isKilling) return;
-
-      // If this is the final task, run the cleanup/completion logic.
+  private checkSyncCompletion = async () => {
+    if (
+      Object.values(this.logFilterProgressTrackers).every((t) =>
+        t.isComplete(),
+      ) &&
+      Object.values(this.factoryChildAddressProgressTrackers).every((t) =>
+        t.isComplete(),
+      ) &&
+      Object.values(this.factoryLogFilterProgressTrackers).every((t) =>
+        t.isComplete(),
+      ) &&
+      this.blockProgressTracker.isComplete()
+    ) {
```

Comment:
Could perhaps extract this into a `private isSyncComplete()` function which could also be used at the end of `start()`.